src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Parameters/SearchParameterSupportResolver.cs

```diff
@@ -86,6 +87,11 @@ public class SearchParameterSupportResolver : ISearchParameterSupportResolver
                 }
             }
 
+            string GetBaseType(string classMapping)
+            {
+                return classMapping.StartsWith(_codeOfTName, StringComparison.Ordinal) ? _codeOfTName : classMapping;
```

Comment:
>_codeOfTName [](start = 89, length = 12)

What does it mean?
I did quick search on hl7.org/fhir and couldn't find that string. Fire.ly repository also show nothing related to it.

Reply:
It means that it is a coded value of a certain type.  Firely includes the details of that type in the string.  For example: "codeOfT<Appointment+status>"  We then have issues trying to look up that value in the mapping dictionary because we use simply "codeOfT" as the key.

Reply:
ClassMappingClass has IsCodeOfT boolean property, I think it's better to use it, than string comparison. 

I also see code like this in firely:
```
     string getFhirTypeName(Type ft)
            {
                var map = PocoStructureDefinitionSummaryProvider.GetMappingForType(ft);
                return map.IsCodeOfT ? "code" : map.Name;
            }
```
maybe we should use "code" instead of "codeOfT", not sure does it matter or not.



---
In reply to: [599083735](https://github.com/microsoft/fhir-server/pull/1761#discussion_r599083735) [](ancestors = 599083735)

Reply:
I updated it to use the boolean value, I agree that is better.  I kept the "codeOfT" string because that is the value loaded into our mapping dictionary.